src/components/date-time-picker/date-time-picker.service.ts

```diff
@@ -52,20 +53,29 @@ export class DateTimePickerService implements OnDestroy {
 
     private _subscription: Subscription;
 
-    constructor(@Optional() private _config: DateTimePickerConfig) {
+    constructor(@Optional() private _config: DateTimePickerConfig,
+    @Optional() rangeService: DateRangeService,
+    @Optional() rangeOptions: DateRangeOptions) {
 
         // when the active date changes set the currently selected date
         this._subscription = this.selected$.subscribe(date => {
-
             // the month and year displayed in the viewport should reflect the newly selected items
             if (date instanceof Date) {
                 this.setViewportMonth(date.getMonth());
                 this.setViewportYear(date.getFullYear());
             }
 
             // emit the new date to the component host but only if they are different
-            if (!dateComparator(date, this.selected$.value)) {
-                this.date$.next(date);
+            if (!dateComparator(date, this.date$.value)) {
```

Comment:
The main issue was that the implementation needed split out based on the use scenario. The `date$` value in the DatePickerService needs checked and updated for time to work on standalone `DateTimePicker` components, whereas this check is not valid for `DateRangePicker` as the `start` and `end` values in `DateRangeService` need to drive date updates which work through to the child `DateTimePicker` components.